api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -251,14 +252,17 @@ public boolean isComplete() {
       Preconditions.checkNotNull(
           expiryDays, "expected value for config key accessRenewal.expiryDays.expiryDays");
       long expiryDaysInMs = TimeUnit.MILLISECONDS.convert(expiryDays, TimeUnit.DAYS);
-      return completion.map(c -> new Timestamp(c.getTime() + expiryDaysInMs));
+      return completion.map(
+          c -> new Timestamp(Math.max(c.getTime() + expiryDaysInMs, minExpirationEpochMs)));
     }
 
-    // IMPORTANT: do not use this method as the sole method of determining access compliance!
-    // We must ALSO confirm module completion.
-    // Modules which are not complete are also "not expired"
```

Comment:
@jmthibault79 FYI.

Reply:
Not sure how I feel about the checkArgument() change.  I would prefer to have a method which is callable under all circumstances.  

But as you point out, the previous state is problematic and we do need to improve it.

When we were designing the `RenewableAccessModuleStatus` we went back and forth about different ideas before we settled on returning `expirationEpochMillis` and `hasExpired` per module.  Now that we have some experience with these concepts, I see that in practice we are always using `hasExpired` to check modules' compliance statuses.  But that's got a subtle difference to it, leading to constructions like below: `times.isComplete() && times.hasExpired()`.

How about this: since what we actually want to know is `isCompliant()` - let's return that in the API instead, and capture all of the semantics about what it means to be compliant (by time) in a ModuleTimes method which replaces `hasExpired()`.

This would require UI changes as well ... and blows up the scope for this PR.  Let's get the 6/30 change in first and come back to this.

Reply:
Sounds good, Joel. I might propose "isCurrent" instead of "isAARCompliant."